### PR TITLE
Add GIT_EMERGECONFLICT and GIT_EDIRECTORY error mapping

### PR DESCRIPTION
**Pull request by odedw**
Added two missing error codes - issue #1203.

### CODE REVIEW

File: LibGit2Sharp/Core/GitErrorCode.cs [hunk 1]
```diff
@@ -107,6 +107,16 @@ internal enum GitErrorCode
...
 
         /// <summary>
+        /// The operation is not valid for a directory.
+        /// </summary>
+        Directory = -23,
+
+        /// <summary>
+        /// A merge conflict exists and cannot continue
+        /// </summary>
+        EmergencyConflict = -24,
```
Comment (arthurschreiber) [inline LibGit2Sharp/Core/GitErrorCode.cs]: Is `EmergencyConflict` a typo? :ambulance:

Comment (odedw) [inline LibGit2Sharp/Core/GitErrorCode.cs]: typo mixed with an oversight:) thanks, fixed.

Comment (nulltoken) [inline LibGit2Sharp/Core/GitErrorCode.cs]: I kind of liked this typo :wink:

**Comment by nulltoken** [comment]
:shipit: :bangbang: 

@odedw Another one in. Congrats and so many thanks :sparkles:

**Comment by odedw** [comment]
@nulltoken Thanks! Glad I can help!

**Pull request merged**